pkg/apis/ingress/v1alpha1/types.go

```diff
@@ -22,8 +22,59 @@ type ClusterIngress struct {
 }
 
 type ClusterIngressSpec struct {
-	// Fill me
+	// IngressDomain is a DNS name suffix serviced by the
+	// ClusterIngress. This value is added automatically to the
+	// status of Routes to inform downstream systems.
+	IngressDomain *string `json:"ingressDomain"`
+
+	// NodePlacement provides additional control over the
+	// scheduling of ingress implementation pods.
+	NodePlacement *NodePlacement `json:"nodePlacement"`
```

Comment:
No reason we couldn't wire this up now—should be as simple as the following:
```go
if cr.Spec.NodePlacement != nil {
    if cr.Spec.NodePlacement.NodeSelector != nil {
        ds.Spec.Template.Spec.NodeSelector = cr.Spec.NodePlacement.NodeSelector
    }
}
```
Right?

Reply:
Well, I guess we'd need some `metav1.LabelSelectorAsSelector` conversion in there as well:
```go
if cr.Spec.NodePlacement != nil {
    if cr.Spec.NodePlacement.NodeSelector != nil {
        nodeSelector, err := metav1.LabelSelectorAsSelector(cr.Spec.RouteSelector)
	if err != nil {
	    return nil, fmt.Errorf("clusteringress %q has invalid spec.routeSelector: %v", cr.Name, err)
	}
        ds.Spec.Template.Spec.NodeSelector = nodeSelector
    }
}
```

Reply:
Please do this in a followup

Reply:
Followup: #39.